### PR TITLE
renamed config key + cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#3083

*Description of changes:*
- renamed VUE_APP_AUTH_ROOT_API to AUTH_API_URL
- renamed API_URL to LEGAL_API_URL
- renamed KEYCLOAK_CONFIG_URL to KEYCLOAK_CONFIG_PATH
- deleted obsolete  SIGNIN_URL
- delete obsolete auth functions, classes, constants and interfaces
- added some missing types
- simplified Vuetify init

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -280,8 +280,8 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
     }
   }
 
-  @Watch('stateModel.nameRequest.entityType')
-  private onEntityTypeChanged (val: string | null) : void {
+  @Watch('state.nameRequest.entityType')
```
Comment (cameron-freshworks) [inline src/App.vue]: I'm not sure this is correct. I had to change the reference to the `stateModel` which we had bound to `App`, as there is no `state` property bound here.

File: src/main.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import 'regenerator-runtime/runtime' // to use transpiled generator functions
 
 // Vue Libraries
 import Vue from 'vue'
-import vuetify from '@/plugins/vuetify'
+import Vuetify from 'vuetify/lib'
```
Comment (cameron-freshworks) [inline src/main.ts]: interesting, what is the difference here?

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue]: Oops, merge issue. Fixing now.

File: src/utils/auth-helpers.ts [hunk 3]
```diff
@@ -1,119 +0,0 @@
...
-  private storeAuthRoles (response: AuthIF): void {
-    const authRoles = response && response.data && response.data.roles
-    if (authRoles && authRoles.length > 0) {
-      // set auth roles to store
-      store.dispatch(SET_AUTH_ROLES, authRoles)
-    } else {
-      throw new Error('Invalid auth roles')
-    }
-  }
-}
```
Comment (cameron-freshworks) [inline src/utils/auth-helpers.ts]: Good clean up! I guess all of this is obsolete with the new kc integration?

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue]: I found another place where my rebase restored some deleted whitespace :/

Comment (cameron-freshworks) [reply src/App.vue]: excellent!

File: src/main.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/main.ts]: We were calling a plugin that did very little so I moved its code here. I think it's cleaner :)

Comment (severinbeauvais) [reply src/main.ts]: That you can't tell the difference proves the plugin did very little :)

File: src/App.vue (hunk 1 above)
Comment (lekshmimallika-aot) [inline src/App.vue]: Should be stateModel I think

File: src/utils/auth-helpers.ts (hunk 3 above)
Comment (severinbeauvais) [reply src/utils/auth-helpers.ts]: Yep!

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue]: Yes, fixed.

**Pull request merged**